src/solver/type_getters.jl

```diff
@@ -135,17 +136,21 @@ function get_spaces(parsed_args, params, comms_ctx)
         warp_function = nothing
     elseif topography == "Earth"
         data_path = joinpath(topo_elev_dataset_path(), "ETOPO1_coarse.nc")
+        array_type = ClimaComms.array_type(ClimaComms.device())
```

Comment:
Why are we getting the device this way? Shouldn't we already have it somewhere?

Reply:
Something like `ctx = ClimaComms.context(axes(Y.c))` to then get `ctx.device` perhaps based on this ```ClimaComms.context(device=device())``` in ClimaComms perhaps ? ; I'll check if we have this stored somewhere already in`model` or `type` getters.

Reply:
@charleskawczynski Fixed (device is accessible in ctx) and is available at this level without calling the ClimaComms `device` function